lms/djangoapps/etc_api/views.py

```diff
@@ -0,0 +1,178 @@
+import json
+import logging
+import string
+import random
+
+from django.core.exceptions import NON_FIELD_ERRORS, ValidationError
+from django.contrib.auth.models import User
+from rest_framework.views import APIView
+from rest_framework import status
+from rest_framework.response import Response
+from util.disable_rate_limit import can_disable_rate_limit
+from openedx.core.djangoapps.user_api.accounts.api import check_account_exists
+from student.views import create_account_with_params
+from student.models import CourseEnrollment, EnrollmentClosedError, \
+    CourseFullError, AlreadyEnrolledError, UserProfile
+from enrollment.views import EnrollmentCrossDomainSessionAuth, \
+    EnrollmentUserThrottle, ApiKeyPermissionMixIn
+from instructor.views.api import save_registration_code, \
+    students_update_enrollment, require_level
+from .serializers import BulkEnrollmentSerializer
+from .utils import  send_activation_email, ApiKeyHeaderPermissionInToken
+from django.contrib.auth import authenticate, login
+from rest_framework.authentication import SessionAuthentication, BasicAuthentication
+from django.core.validators  import validate_slug
+from opaque_keys.edx.locations import SlashSeparatedCourseKey
+
+class CsrfExemptSessionAuthentication(SessionAuthentication):
```

Comment:
Why do you need this class? Could you please explain?

Reply:
This class allows you to turn off checking the security token for the mappings where it is used.